ui/component/settingWalletServer/view.jsx

```diff
@@ -0,0 +1,174 @@
+// @flow
+
+import React, { useState, useEffect } from 'react';
+import { FormField } from 'component/common/form';
+import ServerInputRow from './internal/inputRow';
+import ServerDisplayRow from './internal/displayRow';
+
+type DaemonSettings = {
+  lbryum_servers: Array<>
+};
+
+type StatusOfServer = {
+  host: string,
+  port: string,
+  availability: boolean,
+  latency: number,
+}
+
+type ServerInConfig = Array<string> // ['host', 'port']
+
+type DisplayOfServer = {
+  host: string,
+  port: string,
+  availability: boolean,
+}
+
+type ServerStatus = Array<StatusOfServer>
+type ServerConfig = Array<ServerInConfig>
+type DisplayList = Array<DisplayOfServer>
+
+type Props = {
+  daemonSettings: DaemonSettings,
+  getDaemonStatus: () => void,
+  setWalletServers: any => void,
+  clearWalletServers: () => void,
+  customServers: ServerConfig,
+  saveServers: string => void,
+  fetchDaemonSettings: () => void,
+  serverPrefs: ServerConfig,
+};
+
+function SettingWalletServer(props: Props) {
+  const {
+    daemonSettings,
+    fetchDaemonSettings,
+    setWalletServers,
+    getDaemonStatus,
+    clearWalletServers,
+    saveServers,
+    customServers,
+    serverPrefs,
+  } = props;
+
+  const [custom, setCustom] = useState(false);
+  const [status, setStatus] = useState([]);
+  const serversInConfig: ServerConfig = daemonSettings && daemonSettings.lbryum_servers;
+  const servers = customServers.length ? customServers : serversInConfig;
+  const STATUS_INTERVAL = 5000;
+
+  useEffect(() => {
+    if (serverPrefs && serverPrefs.length) {
+      setCustom(true);
+    }
+  }, []);
+
+  // TODO: do this globally to have status updated for the app
+  useEffect(() => {
+    const interval = setInterval(() => {
+      getDaemonStatus()
+        .then(s => {
+          if (s && s.wallet && s.wallet.servers) {
+            setStatus(s.wallet.servers);
+          }
+        });
+    }, STATUS_INTERVAL);
+    return () => clearInterval(interval);
+  }, []);
+
+  useEffect(() => {
+    fetchDaemonSettings();
+  }, [custom]);
```

Comment:
If you're just fetching this when `custom` changes, maybe just fetch in the onchange handler?

Reply:
Actually, it can just be once []. Because every other action eventually results in DAEMON_SETTINGS_RECEIVED which is the result of fetch and updates daemonSettings.